com.unity.multiplayer.mlapi/Runtime/Metrics/NetworkMetrics.cs

```diff
@@ -38,24 +46,37 @@ public class NetworkMetrics : INetworkMetrics
         private EventMetric<NamedMessageEvent> m_NamedMessageSentEvent = new EventMetric<NamedMessageEvent>("Named Message Sent");
         private EventMetric<NamedMessageEvent> m_NamedMessageReceivedEvent = new EventMetric<NamedMessageEvent>("Named Message Received");
 
+        private Dictionary<ulong, NetworkObjectIdentifier> m_NetworkGameObjects = new Dictionary<ulong, NetworkObjectIdentifier>();
+
+        private ConnectionInfo m_LocalConnectionInfo;
+
         public NetworkMetrics(NetworkManager networkManager)
         {
             m_NetworkManager = networkManager;
             m_Dispatcher = new MetricDispatcherBuilder()
                 .WithMetricEvents(m_NamedMessageSentEvent, m_NamedMessageReceivedEvent)
                 .Build();
-            
+
             m_Dispatcher.RegisterObserver(MLAPIObserver.Observer);
+            m_LocalConnectionInfo = new ConnectionInfo(m_NetworkManager.LocalClientId);
+        }
+
+        public void TrackNetworkObject(NetworkObject networkObject)
+        {
+            if (!m_NetworkGameObjects.ContainsKey(networkObject.NetworkObjectId))
+            {
+                m_NetworkGameObjects[networkObject.NetworkObjectId] = new NetworkObjectIdentifier(networkObject.name, networkObject.NetworkObjectId);
+            }
         }
 
         public void TrackNamedMessageSent(string messageName, ulong bytesCount)
         {
-            m_NamedMessageSentEvent.Mark(new NamedMessageEvent(new ConnectionInfo(m_NetworkManager.LocalClientId), messageName, bytesCount));
+            m_NamedMessageSentEvent.Mark(new NamedMessageEvent(m_LocalConnectionInfo, messageName, bytesCount));
         }
 
         public void TrackNamedMessageReceived(string messageName, ulong bytesCount)
         {
-            m_NamedMessageReceivedEvent.Mark(new NamedMessageEvent(new ConnectionInfo(m_NetworkManager.LocalClientId), messageName, bytesCount));
+            m_NamedMessageReceivedEvent.Mark(new NamedMessageEvent(m_LocalConnectionInfo, messageName, bytesCount));
```

Comment:
I'm confused at why this is `m_LocalConnectionInfo`. Shouldn't connection info be a parameter to this method? Aren't we tracking "I sent a named message to [connection]" or "I received a named message from [connection]"?

Reply:
Same here, when I asked this question JS answered and it made sense for what we want to see in the profiler.

But maybe this is just a question of renaming variables and parameters to make it crystal clear what we're expecting.

Reply:
No. It's the local connection that received a Named Message or sent it.

Reply:
This doesn't make sense to me. Local connection is always involved in the transaction, and it seems pointless to mark that. What's beneficial to the users is to know which other connection is involved in the transaction. This code will cause all data to be grouped under a single connection id which doesn't provide any user value

Reply:
What if you remote profile? The connection will be different than the local one. The current structure that we've agreed doesn't have both connection information. It's always been about the local connection.

Reply:
Remote or not, you always implicitly know that the instance that you're profiling will be involved in the transaction. We don't need both connection information, I'm saying that the connection attached to our data should be the connection that we _don't_ implicitly know about. Attaching the local connection information to the data provides no value to the user because it's information they already implicitly know

Reply:
Here's an example. If you send a client RPC, it sends it to all clients. I would expect that there is an RPCSent event delivered to the profiler _for each connection_ because in actuality the profiled instance just sent N RPCs, not one.

Reply:
After an offline discussion with @bendoyon and @becksebenius-unity . We've aligned that indeed this should be the outgoing and incoming connections. A misunderstanding came from how we currently render the data in the profiler. Ultimately, we are profiling a connection "A" and it's either sending to a connection "B" or receiving it from it and this should reflect that.